pkg/controller/installation/core_controller.go

```diff
@@ -109,33 +110,47 @@ func newReconciler(mgr manager.Manager, opts options.AddOptions) (*ReconcileInst
 
 	statusManager := status.New(mgr.GetClient(), "calico", opts.KubernetesVersion)
 
-	// The typhaAutoscaler needs a clientset.
+	// The typhaAutoscaler and calicoWindowsUpgrader need a clientset.
 	cs, err := kubernetes.NewForConfig(mgr.GetConfig())
 	if err != nil {
 		return nil, err
 	}
 
-	// Create a Typha autoscaler.
+	// Create the SharedIndexInformer used by the typhaAutoscaler and
+	// calicoWindowsUpgrader.
 	nodeListWatch := cache.NewListWatchFromClient(cs.CoreV1().RESTClient(), "nodes", "", fields.Everything())
+	nodeIndexInformer := cache.NewSharedIndexInformer(nodeListWatch, &corev1.Node{}, 0, cache.Indexers{})
+
+	go nodeIndexInformer.Run(opts.ShutdownContext.Done())
+	for nodeIndexInformer.HasSynced() {
```

Comment:
I think we should do this check within the typha autoscaler and windows upgrader, rather than block the entire reconcile here

Reply:
I don't think the rest of the core controller needs to depend on this syncing, right?

Reply:
Addressed with fd2289ef9ee93d5362ae06be843883f725fcdf91